src/tf/tf.cpp

```diff
@@ -939,23 +1033,43 @@ struct tf_parser
                     continue;
                 if(nodes.count(input) > 0)
                 {
-                    auto&& iname = get_name(nodes.at(input));
-                    assert(name != iname);
-                    this->parse_node(iname);
-                    args.push_back(instructions.at(iname));
+                    std::string iname;
+                    // input was from a node with multiple outputs
+                    if(contains(input, ':'))
+                    {
+                        auto orig_iname = input.substr(0, input.find(':'));
+                        assert(name != orig_iname);
+                        this->parse_node(orig_iname);
+                        args.push_back(instructions.at(input));
+                    }
+                    else
+                    {
+                        iname = get_name(nodes.at(input));
```

Comment:
Only the `iname` assignment should be in the `if`:

```cpp
                    if(contains(input, ':'))
                    {
                        iname = input.substr(0, input.find(':'));
                    }
                    else
                    {
                        iname = get_name(nodes.at(input));
                    }
                    assert(name != iname);
                    this->parse_node(iname);
                    args.push_back(instructions.at(iname));
```

Reply:
fixed